students/philip_behrend/lesson05/mailroom_pt3.py

```diff
@@ -77,25 +88,5 @@ def quit_program(donor_dict):
         arg_dict = {1: send_thanks, 2: create_report, 3: send_all_letters, 4: quit_program}
         response = arg_dict.get(response)(donor_dict)
```

Comment:
I felt like a while loop here would be a cleaner implementation (simply re-display the response options if response not in [1,2,3,4[). Even though it doesn't use the try / except nomenclature, it still handles the exception by prompting the user again. Is that alright?

Reply:
in real code yes, you get to choose either to use exception handling or checking via if statements. typically, you would choose exception handling if it's something exceptional.

in this particular case I called it out because lesson 05 is about utilizing exception handling in your mailroom code for learning purposes and to see how your code changes 🙂 

In the mailroom, I actually think exception handling makes for a cleaner code. Check this out:
```
while True:
    response = input('...')
    try:
        switch_dict[response]()
    except keyError:
        print("invalid option!")
```
```